Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1

```diff
@@ -216,8 +222,8 @@ function HasElements
 #
 function IsPrBuild
 {    
-    $sourceBranch = $env:Build_SourceBranch
-    $scProvider = $env:Build_Repository_Provider
+    $sourceBranch = GetTaskContextVariable "Build.SourceBranch"
+    $scProvider = GetTaskContextVariable "Build.Repository.Provider" 
```

Comment:
No point calling this if not 
$sourceBranch -and `
$sourceBranch.StartsWith("refs/pull/", [StringComparison]::OrdinalIgnoreCase)        		              
So you probably want to break the return into two conditions...